qoajad_backend/src/main/java/com/qoajad/backend/controller/DefaultController.java

```diff
@@ -0,0 +1,93 @@
+package com.qoajad.backend.controller;
+
+import com.qoajad.backend.model.User;
+import com.qoajad.backend.service.user.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.cloud.context.config.annotation.RefreshScope;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import java.util.List;
+
+@RefreshScope
+@RestController
+public class DefaultController {
+
+    private final UserService userService;
+
+    @Autowired
+    public DefaultController(@Qualifier("defaultUserService") final UserService userService) {
+        this.userService = userService;
+    }
+
+    @RequestMapping(value = "/users", method = RequestMethod.GET)
+    public ResponseEntity<List<User>> retrieveAllUsers() {
+        ResponseEntity<List<User>> response;
+        try {
+            response = new ResponseEntity<>(userService.retrieveAllUsers(), HttpStatus.OK);
+        } catch (EmptyResultDataAccessException e) {
+            response = new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        } catch (DataIntegrityViolationException e) {
+            response = new ResponseEntity<>(HttpStatus.CONFLICT);
+        }
+        return response;
+    }
+
+    @RequestMapping(value = "/user/{id}", method = RequestMethod.GET)
+    public ResponseEntity<User> readUser(@PathVariable int id) {
```

Comment:
I suggest you to rename this and downstream code for a more expressive name. findUserById perhaps? findUser? retrieveUser?, Read user can be misleading